L1Trigger/L1CaloTrigger/test/generatePatternPF.py

```diff
@@ -0,0 +1,174 @@
+
+import os
+
+# options
+import FWCore.ParameterSet.VarParsing as VarParsing
+options = VarParsing.VarParsing('analysis')
+options.register('filename',
+                 "rx_summary",
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.string,
+                 "Filename string")
+options.register('outDir',
+                 "rxFiles",
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.string,
+                 "Output directory for buffer files")
+options.register('nPayloadFrames',
+                 13,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.int,
+                 "N payload frames per event")
+options.register('nHeaderFrames',
+                 0,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.int,
+                 "N header frames per event")
+options.register('nClearFrames',
+                 0,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.int,
+                 "N clear frames between events")
+options.register('dump',
+                 True,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.bool,
+                 "Dump messages")
+options.register('pattern',
+                 True,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.bool,
+                 "Save pattern file")
+options.register('edm',
+                 False,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.bool,
+                 "Save EDM file")
+options.register('debug',
+                 False,
+                 VarParsing.VarParsing.multiplicity.singleton,
+                 VarParsing.VarParsing.varType.bool,
+                 "Enable debug data")
+#options.register('doMP',
+#                 True,
+#                 VarParsing.VarParsing.multiplicity.singleton,
+#                 VarParsing.VarParsing.varType.bool,
+#                 "Generate MP pattern")
+#options.register('doDemux',
+#                 True,
+#                 VarParsing.VarParsing.multiplicity.singleton,
+#                 VarParsing.VarParsing.varType.bool,
+#                 "Read demux data")
+#options.register('nMP',
+#                 1,
+#                 VarParsing.VarParsing.multiplicity.singleton,
+#                 VarParsing.VarParsing.varType.int,
+#                 "Number of MPs")
+
+options.parseArguments()
+
+if (options.maxEvents == -1):
+    options.maxEvents = 1
+
+fileList = FileUtils.loadListFromFile('ttbar.list')
+readFiles = cms.untracked.vstring(*fileList)
+
+# make output directory if it doesn't already exist
+try:
```

Comment:
try/except is not needed here:
```python
if not os.path.isdir(options.outDir): os.mkdir(options.outDir)
```

Reply:
fixed: https://github.com/cms-l1t-offline/cmssw/blob/fc3a1a454f40efdf7c95dd97295a6a085a674092/L1Trigger/L1CaloTrigger/test/generatePatternPF.py#L70